libraries/AP_HAL_Linux/system.cpp

```diff
@@ -27,7 +27,9 @@ void panic(const char *errormsg, ...)
     va_start(ap, errormsg);
     vdprintf(1, errormsg, ap);
     va_end(ap);
-    write(1, "\n", 1);
+    if (write(1, "\n", 1) != 1) {
```

Comment:
I think a more correct approach is using pragmas:
```c++
    #pragma GCC diagnostic push
    #pragma GCC diagnostic ignored "-Wunused-result"
    write(1, "\n", 1);
    #pragma GCC diagnostic pop
```

... and it works for both GCC and clang. I know that is more verbose than using `if`, but I think it's makes more sense.